### PR TITLE
[config] Added nullness annotations and suppress 'ConfigStatusInfoEvent' if list of 'ConfigStatusMessage's is empty

### PR DESCRIPTION
**Pull request by cweitkamp**
- Added nullness annotations
- Use constructor injection
- Suppress `ConfigStatusInfoEvent` if list of `ConfigStatusMessage`s is empty

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.core.config.core/src/test/java/org/openhab/core/config/core/status/ConfigStatusInfoTest.java [hunk 1]
```diff
@@ -90,17 +87,17 @@ public void assertCorrectConfigErrorHandlingUsingAddConfigError() {
...
+    // public void assertNPEisThrownIfTypesAreNull() {
+    // ConfigStatusInfo info = new ConfigStatusInfo();
+    // assertThrows(NullPointerException.class, () -> info.getConfigStatusMessages(null, Set.of()));
+    // }
+    //
+    // @Test
+    // public void assertNPEisThrownIfParameterNamesAreNull() {
+    // ConfigStatusInfo info = new ConfigStatusInfo();
+    // assertThrows(NullPointerException.class, () -> info.getConfigStatusMessages(Set.of(), null));
+    // }
```
Comment (wborn) [inline bundles/org.openhab.core.config.core/src/test/java/org/openhab/core/config/core/status/ConfigStatusInfoTest.java:100]: I think such tests are quite useless when the classes are annotated.

```suggestion
```

**Review by wborn — COMMENTED** [review]
Thanks for making these improvements! I have one comment.

File: bundles/org.openhab.core.config.core/src/test/java/org/openhab/core/config/core/status/ConfigStatusInfoTest.java (hunk 1 above)
Comment (cweitkamp) [reply bundles/org.openhab.core.config.core/src/test/java/org/openhab/core/config/core/status/ConfigStatusInfoTest.java]: Sure. Done.

**Review by wborn — APPROVED** [review]
Thank you!

**Pull request merged**